AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -109,46 +108,53 @@ - (void)setParent:(id<ASLayoutable>)parent
 
 - (void)setChild:(id<ASLayoutable>)child;
 {
-  [self setChild:child forIdentifier:kDefaultChildKey];
+  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  
+  id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
+  _child = finalLayoutable;
+  [self propagateUpLayoutable:finalLayoutable];
 }
 
 - (void)setChild:(id<ASLayoutable>)child forIdentifier:(NSString *)identifier
 {
   ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  
   id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
-  self.layoutChildren[identifier] = finalLayoutable;
-  if ([finalLayoutable isKindOfClass:[ASLayoutSpec class]]) {
-    [(ASLayoutSpec *)finalLayoutable setParent:self]; // This will trigger upward propogation if needed.
-  } else if ([self supportsUpwardPropagation]) {
-    ASEnvironmentStatePropagateUp(self, finalLayoutable.environmentState.layoutOptionsState); // Probably an ASDisplayNode
-  }
+  self.childrenWithIdentifier[identifier] = finalLayoutable;
+  
+  // TODO: Should we propagate up the layoutable at it could happen that multiple children will propagated up their
+  //       layout options and one child will overwrite values from another child
+  // [self propagateUpLayoutable:finalLayoutable];
```

Comment:
@maicki good question, we will definitely revisit this in more detail once we aim to re-enable propagation.

I do think in some cases like a vertical stack of two text nodes with flexShrink = YES, we may even want to propagate up from multiple chilrden.  But before we do this stuff we need to go back to the whiteboard and really think carefully about the exact situations it makes sense, and then add them to documentation.